SDK_dotnet/Clients/GluwaClient.cs

```diff
@@ -392,6 +426,100 @@ private string getTimestampSignature(string privateKey)
+            {
+                throw new ArgumentNullException(nameof(privateKey));
+            }
+
+            BitcoinSecret secret = new BitcoinSecret(privateKey, mNetwork);
+            string Timestamp = new DateTimeOffset(DateTime.UtcNow).ToUnixTimeSeconds().ToString();
+            string signature = secret.PrivateKey.SignMessage(Timestamp);
+
+            string signatureToEncode = $"{Timestamp}.{signature}";
+            byte[] signatureByte = Encoding.UTF8.GetBytes(signatureToEncode);
+            string encodedData = Convert.ToBase64String(signatureByte);
+
+            return encodedData;
+        }
+
+        private string getGluwaCoinSignature(ECurrency currency, string amount, string fee, string nonce, string address, string target, string privateKey)
+        {
+            BigInteger convertAmount = GluwacoinConverter.ConvertToGluwacoinBigInteger(amount);
+            BigInteger convertFee = GluwacoinConverter.ConvertToGluwacoinBigInteger(fee.ToString());
+
+            ABIEncode abiEncode = new ABIEncode();
+            byte[] messageHash = abiEncode.GetSha3ABIEncodedPacked(
+                new ABIValue("address", getContractAddress(currency)),
+                new ABIValue("address", address),
+                new ABIValue("address", target),
+                new ABIValue("uint256", convertAmount),
+                new ABIValue("uint256", convertFee),
+                new ABIValue("uint256", int.Parse(nonce))
+                );
+
+            EthereumMessageSigner signer = new EthereumMessageSigner();
+            string signature = signer.Sign(messageHash, privateKey);
+
+            return signature;
+        }
+
+        private async Task<string> getBtcSignatureAsync(ECurrency currency, string address, string amount, string fee, string target, string privateKey)
+        {
+            Result<BalanceResponse, ErrorResponse> getUspentOutput = await GetBalanceAsync(currency, address, true);
+            List<UnspentOutput> unspentOutputs = getUspentOutput.Data.UnspentOutputs.OrderByDescending(u => u.Amount).ToList();
+
+            Money amountValue = Money.Parse(amount);
+            Money feeValue = Money.Parse(fee);
+            Money totalAmountAndFeeValue = amountValue + feeValue;
+            BigInteger totalAmountAndFee = new BigInteger(totalAmountAndFeeValue.Satoshi);
+
+            BitcoinAddress sourceAddress = BitcoinAddress.Create(address, mNetwork);
+            BitcoinAddress targetAddress = BitcoinAddress.Create(target, mNetwork);
+            BitcoinSecret secret = new BitcoinSecret(privateKey, mNetwork);
+
+            List<UnspentOutput> usingUnspentOutputs = new List<UnspentOutput>();
+            BigInteger unspentOutputTotalAmount = BigInteger.Zero;
+            for (int i = 0; i < unspentOutputs.Count; i++)
+            {
+                if (totalAmountAndFee > unspentOutputTotalAmount)
+                {
+                    usingUnspentOutputs.Add(unspentOutputs[i]);
```

Comment:
You should not use too many unspent outputs for one transaction, because fee becomes more expensive when you have more outputs, and the transaction will take too long to be confirmed. 

I suggest adding a constant for maximum output count (like 5) and return an error when you have to use more outputs than that.